### ISSUE TITLE
Normalize Pose, Normalize Face and Draw Landmarks nodes: drive a trained performer with a different body

### ISSUE THREAD

**Issue by fdb** [figmentapp/figment#106]
## Summary

Drive a pose-to-image model that was trained on one body with the skeleton of a different body. Two new nodes, `Normalize Pose` and `Normalize Face`, scale and translate detected landmarks so their size and position match the values the model was trained on. The user types those reference values in; the node measures the live person on its own, in a configurable way. A third node, `Draw Landmarks`, turns the transformed landmarks back into an image.

This is the "global pose normalization" step of Everybody Dance Now (Chan et al. 2019). Follows figmentapp/figment#104 (per-limb colouring) and figmentapp/figment#105 (landmark smoothing).

## The idea in one paragraph

The model only ever saw the performer at one size on screen, with the feet on one line. A guest is taller or shorter and stands closer or further away, so their skeleton lands somewhere the model has never seen. The fix measures two numbers on the guest, the anchor (where the feet are) and the size (feet to nose), zooms the whole skeleton about the anchor until the size matches the performer's, then slides the anchor onto the performer's anchor. Uniform zoom only, never a stretch, so the guest keeps their own proportions.

## Units

All values are in Figment's landmark units: fractions of the image, 0 to 1, origin top left, y down. That is what Detect Pose and Detect Faces output, and it is independent of resolution: a reference floor of `0.9` is row 461 of a 512 px training image and row 972 of a 1080p live frame. `z` is in the same scale as `x` and is multiplied by the same zoom factor. Never use pixels anywhere in these nodes.

## Where the reference values come from

The training set contains skeleton drawings, not landmarks, so the training scripts cannot measure the performer. Measure in Figment instead, with the same detector that made the drawings:

1. `Load Movie` (the performer footage) or `Load Image Folder` (the extracted frames) → `Detect Pose` → `Normalize Pose`.
2. Read the node's **measured floor**, **measured height** and **measured x** outputs after a few seconds (use `measure: on reset` with a long window, or `continuous`).
3. Type those numbers into **reference floor**, **reference height** and **reference x** of the `Normalize Pose` node in the live graph.

The dataset graph itself must not contain a normalize node: the performer must be saved exactly where the camera saw them. The measurement graph above is a separate, throwaway graph.

## Scope

- `src/landmark-normalize.js`: pure functions and the estimator, with unit tests.
- `src/nodes/ml/normalizePose.js`, `src/nodes/ml/normalizeFace.js`, `src/nodes/ml/drawLandmarks.js`.
- Docs: one page per node, plus a section in the pix2pix tutorial.

Non-goals: rotation (head roll or body lean stay as they are; they are part of the performance), hands, Rokoko input, per-limb retargeting, writing landmarks to disk.

### 1. Shared module (`src/landmark-normalize.js`)

Three pure pieces. No node code, no GPU, no timers inside; the node passes the time in.

**Measurement recipes.** Each returns `{ anchor: { x, y }, size }` in landmark units, or `null` when a needed landmark is missing.

| Recipe | Anchor | Size |
|---|---|---|
| `measurePose(landmarks)` | `x`: midpoint of the hips (BlazePose 23, 24); `y`: midpoint of the ankles (27, 28) | vertical distance from the ankle line to the nose (0): `anchor.y - nose.y` |
| `measureFace(landmarks)` | midpoint of the outer eye corners (Face Mesh 33, 263) | Euclidean distance between those two corners |

Vertical distance for the pose, not Euclidean, so raised arms and a bent torso do not change it; a crouch does, which is why the estimator below averages over time. The horizontal anchor is the hips, not the ankles: a step forward moves one ankle and would shift the whole body if the ankle midpoint were the anchor. The hips are the body's root, which is also what Rokoko locks in its treadmill mode.

**Transform.** `normalizeLandmarks(landmarks, frame, estimate, reference, { horizontal })` returns a new array. `frame` is this frame's own measurement, `estimate` is the estimator's current value, `reference` is what the user typed in. With `s = reference.size / estimate.size`:

```
y' = reference.anchor.y + (y - estimate.anchor.y) * s
x' = cx + (x - ax) * s
z' = z * s
```

where `horizontal` picks the horizontal anchor `ax` and the destination `cx`:

| `horizontal` | `ax` (scale about) | `cx` (lands on) | Effect |
|---|---|---|---|
| `keep` | `estimate.anchor.x` | `estimate.anchor.x` | Zoom only. Walking left and right shows as it is. |
| `follow` | `estimate.anchor.x` | `reference.anchor.x` | The person's usual position over the window maps to the reference x. Steps and sways still show; a slow drift across the stage is removed. |
| `treadmill` | `frame.anchor.x` | `reference.anchor.x` | The hips sit on the reference x on every frame. All sideways travel is removed, like Rokoko's treadmill mode; the legs still step, the body stays put. |

`treadmill` uses the per-frame anchor, not the estimate, so it must not pass through the estimator's window. Vertical stays on the estimate in every mode, so jumps survive.

`visibility` and any other fields copy through unchanged. When `estimate.size` is 0, or `estimate` or `frame` is `null`, return the input unchanged.

**Estimator.** `class MeasurementEstimator({ mode, windowSeconds })` with `push(measurement, tSeconds)`, `current()`, `reset()`:

- `manual`: `current()` returns whatever `setManual({ anchor, size })` was given; `push` is ignored.
- `continuous`: keeps the samples of the last `windowSeconds`. `current()` is the median of `size`, the median of `anchor.x`, and for `anchor.y` the 90th percentile (the lowest the feet get is the standing floor; a jump moves the feet up, never down). `windowSeconds` is the knob the user turns: short follows the person as they walk toward the camera, long keeps jumps and crouches.
- `on reset`: like `continuous` during the first `windowSeconds` after `reset()`, then frozen. Before the window has one full second of samples, `current()` returns the running estimate anyway, so the node never outputs nothing.

A sample count on the order of 30 fps × 30 s is small; a sorted copy per frame is fine. Do not add a dependency for percentiles.

### 2. `Normalize Pose` node (`src/nodes/ml/normalizePose.js`)

Ports in:

- `landmarks` object: the `{ type: 'pose', landmarks: [...] }` object from Detect Pose. Other `type` values pass through untouched with a node error.
- **reference floor** number, 0–1, default `0.9`: the performer's ankle line in the training frames.
- **reference height** number, 0–1, default `0.6`: the performer's ankles-to-nose height in the training frames.
- **reference x** number, 0–1, default `0.5`: where the hips land in `follow` and `treadmill`.
- **horizontal** select `keep` / `follow` / `treadmill`, default `keep`: see the transform table. `keep`, because where someone stands left to right is usually part of the performance; `treadmill` when the show wants the performer pinned to one spot whatever the guest does.
- **measure** select `continuous` / `on reset` / `manual`, default `continuous`.
- **window** number in seconds, 0.5–30, default `3`.
- **driver floor**, **driver height**, **driver x** numbers, 0–1: only read in `manual` mode.

Ports out:

- `landmarks` object: same shape, transformed.
- **measured floor**, **measured height**, **measured x** numbers: the estimator's current values for the first person. These are also how a user reads the reference values off the performer footage (see above).

Behaviour:

- One estimator per person slot (index in the `landmarks` array). A slot that disappears for more than `window` seconds is reset.
- `node.onReset` resets every estimator; that is the "stand still for a moment" gesture in `on reset` mode.
- Time base: `performance.now() / 1000` once per render.
- Changing **measure** or **window** resets the estimators.

### 3. `Normalize Face` node (`src/nodes/ml/normalizeFace.js`)

Same node, with the face recipe and face words:

- Accepts `{ type: 'face', landmarks: [...] }` from Detect Faces.
- **reference eye center x**, **reference eye center y** (defaults `0.5`, `0.4`) and **reference eye distance** (default `0.25`).
- **horizontal** defaults to `treadmill`: a face-driven model is usually trained on a centred face, and the guest's face wanders. The per-frame anchor is the eye centre.
- Outputs **measured eye center x**, **measured eye center y**, **measured eye distance**.
- The estimator uses the median for all three; there is no floor rule for faces.

Share the node body between the two nodes through a small helper in `src/landmark-normalize.js` (`createNormalizeNode(node, figment, recipe)` or similar), so the estimator wiring exists once.

### 4. `Draw Landmarks` node (`src/nodes/ml/drawLandmarks.js`)

- `landmarks` object in; **width**, **height** numbers (defaults 512, 512); the drawing parameters Detect Pose has: background, coloring (`solid` / `openpose`, #104), draw points, points color and radius, draw lines, lines color and width.
- Image out, drawn with `LandmarkRenderer`.
- The connection table follows `type`: `pose` → `POSE_CONNECTIONS`, `hand` → `HAND_CONNECTIONS`, `face` → `FACE_LANDMARKS_CONTOURS` (tesselation is Detect Faces' business, keep contours here). `openpose` coloring applies to `pose` only; other types fall back to `solid`.
- If #104 extracted the drawing parameter block of Detect Pose into a shared helper, use it here; if not, do that extraction in this issue so the two nodes cannot drift apart. Detect Pose keeps drawing itself; existing projects do not change.

### 5. Docs

- `docs/src/content/docs/nodes/normalize-pose.md`, `normalize-face.md`, `draw-landmarks.md`, in the style of `detect-pose.md`.
- In `docs/src/content/docs/tutorials/pix2pix.md`, a section "Driving the model with a different body": the live graph `Webcam → Detect Pose → Normalize Pose → Draw Landmarks → ONNX Image Model`, the measurement graph to read the reference values off the performer footage, and the sentence that the dataset graph must not contain a normalize node. State the units once.

## Tests (Vitest, red/green, one at a time)

`src/landmark-normalize.test.js`:

1. `measurePose`: anchor is the ankle midpoint, size is the vertical ankle-to-nose distance, arm positions do not change it.
2. `measureFace`: anchor is the midpoint of 33 and 263, size is their distance.
3. `normalizeLandmarks`: identity when the estimate equals the reference; a taller estimated size shrinks about the anchor; the ankle line lands on the reference floor; `z` scales by the same factor; `visibility` copies through.
3b. Horizontal modes: `keep` leaves `x` untranslated; `follow` puts the estimated hip x on the reference x while a per-frame offset from it still shows; `treadmill` puts this frame's hip x on the reference x exactly, on every frame.
4. Estimator `continuous`: one outlier frame does not move the estimate; a sustained change does after `windowSeconds`; the floor estimate takes the lowest standing position, not the mid-jump one.
5. Estimator `on reset`: frozen after the window; `reset()` re-opens it.
6. Estimator `manual`: returns the manual values and ignores `push`.

`Draw Landmarks`: with `openpose` coloring, the `OverlayBatch` instance list equals Detect Pose's for the same landmarks and parameters.

## Acceptance criteria

- [ ] Three new nodes with docs.
- [ ] A guest of a different height, driving a model trained on the reference body, produces a skeleton whose ankle line and head height sit where the training subject's did, with body proportions unchanged.
- [ ] A jump stays a jump with `window` at 3 s or more in `continuous` mode, and always in `on reset` mode.
- [ ] In `treadmill` mode a guest walking across the stage produces a skeleton that stays on the reference x, with legs still stepping and a jump still visible.
- [ ] The measured outputs of `Normalize Pose`, fed with the performer footage, give numbers that can be typed straight into the reference inputs.
- [ ] `Normalize Face` does the same for Detect Faces output.
- [ ] Detect Pose and Detect Faces rendering is unchanged for existing projects.
- [ ] `npm run format`, `npm test` and `npm run build` green.

## Notes for the implementer

- Everything here is CPU work on about 1 KB of landmarks per frame. No GPU readback, no `await` in the render path.
- Keep the recipes, the transform and the estimator free of node and Figment imports so the tests need no GPU stubs.
- The reference inputs are ordinary number ports, so they can also be driven by expressions or OSC; do not special-case them.
